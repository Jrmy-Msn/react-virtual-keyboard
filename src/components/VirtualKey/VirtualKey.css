.VirtualKey {
  background-color: var(--primary);
  color: var(--primary-text-light);
  padding: .5rem;
  width: var(--letter-width);
  height: 1.5rem;
  display: flex;
  justify-content: center;
  align-items: center;
  user-select: none;
  font-weight: bold;
}

.VirtualKey.pressed {
  cursor: pointer;
  background-color: var(--primary-light);
  color: var(--primary-text-dark);
}
