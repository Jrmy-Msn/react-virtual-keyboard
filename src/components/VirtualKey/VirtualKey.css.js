import {makeStyles} from "@material-ui/core";

export const useStyles = makeStyles(theme => ({
  root: {
    borderRight: 'none !important',
    borderRadius: 0,
    minWidth: 'inherit',
  },
  active({active}) {
    return active ? {
      backgroundColor: theme.palette.primary.dark
    } : {}
  }
}))