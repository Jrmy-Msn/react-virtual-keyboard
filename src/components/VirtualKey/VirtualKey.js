import React from 'react'
import PropTypes from 'prop-types'

import {useStyles} from "./VirtualKey.css.js";
import {Button} from "@material-ui/core";

const TAG = '[VirtualKey]'

const VirtualKey = ({value, active, onClick, onMouseOver, onMouseOut, ...other}) => {
  const classes = useStyles({active});
  return (
    <Button
      {...other}
      color={'primary'}
      variant={'contained'}
      className={`VirtualKey ${other.className} ${classes.active}`}
      aria-label={`${typeof value === 'string' ? value : ''}`}
      classes={{root: classes.root}}
      onMouseOut={() => onMouseOut(value)}
      onMouseOver={() => onMouseOver(value)}
      onClick={() => onClick(value)}>{value}</Button>
  )
}

VirtualKey.propTypes = {
  value: PropTypes.any.isRequired,
  feedback: PropTypes.string,
  onClick: PropTypes.func,
  onMouseOver: PropTypes.func,
  onMouseOut: PropTypes.func,
}

VirtualKey.defaultProps = {
  onClick: () => void undefined,
  onMouseOver: () => void undefined,
  onMouseOut: () => void undefined,
  feedback: '',
}


export default VirtualKey