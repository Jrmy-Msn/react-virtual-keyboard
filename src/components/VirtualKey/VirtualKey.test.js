import React from 'react'
import {cleanup, fireEvent, render, screen} from '@testing-library/react'

import VirtualKey from "../VirtualKey/VirtualKey"

afterEach(cleanup)

it('should display "value" prop', () => {
  render(<VirtualKey value="J"/>)
  expect(screen.getByRole('button')).not.toBeNull()
})

it('should add the "feedback" prop CSS class', () => {
  let {container} = render(<VirtualKey feedback="pressed" value="E"/>)
  expect(screen.getByRole('button')).toHaveClass('pressed')
})

it('should call "onClick" prop, after click, with "value" prop as argument', () => {
  const onClick = jest.fn()
  render(<VirtualKey value="M" onClick={onClick}/>)
  fireEvent.click(screen.getByRole('button'))
  expect(onClick).toHaveBeenCalledTimes(1)
  expect(onClick).toHaveBeenCalledWith("M")
})

it('should call "onMouseOver" prop, when mouse over, with "value" prop as argument', () => {
  const onMouseOver = jest.fn()
  render(<VirtualKey value="M" onMouseOver={onMouseOver}/>)
  fireEvent.mouseOver(screen.getByRole('button'))
  expect(onMouseOver).toHaveBeenCalledTimes(1)
  expect(onMouseOver).toHaveBeenCalledWith("M")
})

it('should call "onMouseOut" prop, when mouse over, with "value" prop as argument', () => {
  const onMouseOut = jest.fn()
  render(<VirtualKey value="M" onMouseOut={onMouseOut}/>)
  fireEvent.mouseOut(screen.getByRole('button'))
  expect(onMouseOut).toHaveBeenCalledTimes(1)
  expect(onMouseOut).toHaveBeenCalledWith("M")
})

