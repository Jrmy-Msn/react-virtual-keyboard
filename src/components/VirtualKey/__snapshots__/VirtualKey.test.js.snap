// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VirtualKey Component Render renders correctly 1`] = `
<button
  aria-label="M"
  className="MuiButtonBase-root MuiButton-root makeStyles-root-1 MuiButton-contained VirtualKey undefined makeStyles-active-2 makeStyles-active-10 MuiButton-containedPrimary"
  disabled={false}
  feedback=""
  onBlur={[Function]}
  onClick={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseOut={[Function]}
  onMouseOver={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  <span
    className="MuiButton-label"
  >
    M
  </span>
</button>
`;
